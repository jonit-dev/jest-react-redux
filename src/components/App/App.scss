.app-container {
  margin-top: 10rem;
  min-height: 100vh;
}

.app-list {
  position: relative;
}
.app-list .fa-trash {
  position: absolute;
  top: 1rem;
  right: 4rem;
  color: darkslategrey;
  opacity: 0;
  transition: all 0.2s ease-in-out;
}

.app-list:hover .fa-trash {
  opacity: 1;
  cursor: pointer;
}
