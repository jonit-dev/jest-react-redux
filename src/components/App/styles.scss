@import "./App.scss";
@import "../GiftForm/GiftForm.scss";
