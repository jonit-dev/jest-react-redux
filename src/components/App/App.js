import React, { Component } from "react";
import Header from "../Header";
import "./styles.scss";
import GiftForm from "../GiftForm/GiftForm";

class App extends Component {
  render() {
    return (
      <>
        <Header />
        <div className="container app-container">
          <div className="col-md-12">
            <div className="jumbotron" data-test="jumbotron">
              <h2>Add your gift</h2>
              <p>Write down your gift and the click the button below</p>

              <GiftForm />
            </div>
          </div>
        </div>
      </>
    );
  }
}

export default App;
