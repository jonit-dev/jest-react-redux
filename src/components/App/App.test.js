import React from "react";
import { shallow } from "enzyme";
import App from "./App";

const renderApp = (props = {}) => {
  const app = shallow(<App {...props} />);
  return app;
};

describe("check App component basic functionality", () => {
  let app;
  beforeEach(() => {
    //setup app component properly, before each test
    app = renderApp();
  });
  it("should render correctly", () => {
    expect(app).toMatchSnapshot();
  });

  it("jumbotron should load correctly", () => {
    const jumbotron = app.find("[data-test='jumbotron']");

    //you can debug a component code
    // console.log(jumbotron.debug());

    expect(jumbotron.length).toBe(1);
  });
});
