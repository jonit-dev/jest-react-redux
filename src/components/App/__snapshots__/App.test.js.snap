// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check App component basic functionality should render correctly 1`] = `ShallowWrapper {}`;
